semantic/check_test.go

```diff
@@ -1216,3 +1338,960 @@ func TestVarAssignmentChecks(t *testing.T) {
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "a"},
+														Init:       &IntegerLiteral{Value: 1},
+													},
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "g"},
+														Init: &FunctionExpression{
+															Block: &FunctionBlock{
+																Parameters: &FunctionParameters{
+																	List: []*FunctionParameter{
+																		{
+																			Key: &Identifier{Name: "g"},
+																		},
+																	},
+																},
+																Body: &CallExpression{
+																	Callee: &IdentifierExpression{Name: "f"},
+																	Pipe:   &IdentifierExpression{Name: "g"},
+																},
+															},
+														},
+													},
+													&NativeVariableAssignment{
+														Identifier: &Identifier{Name: "h"},
+														Init: &FunctionExpression{
+															Block: &FunctionBlock{
+																Parameters: &FunctionParameters{
+																	List: []*FunctionParameter{
+																		{
+																			Key: &Identifier{Name: "b"},
+																		},
+																	},
+																},
+																Body: &CallExpression{
+																	Callee: &IdentifierExpression{Name: "g"},
+																	Arguments: &ObjectExpression{
+																		Properties: []*Property{
+																			{
+																				Key:   &Identifier{Name: "g"},
+																				Value: &IdentifierExpression{Name: "b"},
+																			},
+																		},
+																	},
+																},
+															},
+														},
+													},
+													&ReturnStatement{
+														Argument: &CallExpression{
+															Callee: &IdentifierExpression{Name: "h"},
+															Arguments: &ObjectExpression{
+																Properties: []*Property{
+																	{
+																		Key:   &Identifier{Name: "b"},
+																		Value: &IdentifierExpression{Name: "a"},
+																	},
+																},
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					{
+						Package: &PackageClause{
+							Name: &Identifier{Name: "foo"},
+						},
+						Body: []Statement{
+							&OptionStatement{
+								Assignment: &NativeVariableAssignment{
+									Identifier: &Identifier{Name: "b"},
+									Init: &IdentifierExpression{
+										loc: loc{
+											File: "b.flux",
+											Start: ast.Position{
+												Line:   2,
+												Column: 12,
+											},
+											End: ast.Position{
+												Line:   2,
+												Column: 13,
+											},
+										},
+										Name: "a",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			err: fmt.Errorf(`option dependency: option "b" depends on option "a" defined in the same package at b.flux|2:12-2:13`),
+		},
+	}
```

Comment:
Is it possible to have cyclical dependencies?
```
package p
option o1 = r.o2
```
```
package r
option o2 = p.o1
```

Reply:
I think your walker will catch this, but it would be good to have a test.

Reply:
This walker will only check option dependencies across a single package. Cyclic dependencies between packages would require a different method.

Reply:
I guess my example doesn't have any `import` statements, so my question is, do we check for cyclical imports?

Reply:
We do not check for cyclic imports at the moment but it is needed per the spec. I don't believe there is an issue for it so I will create it.